src/hooks/useSearchParams.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+
+/**
+ * Take a url string and convert it to a search params map
+ *
+ * @param {string} search params url string
+ * @returns {{}} a key:value map of the search params
+ */
+const toMap = (search: string) => {
+  const params = new URLSearchParams(search);
+  const map = {};
+  params.forEach((value, key) => {
+    map[key] = value;
+  });
+  return map;
+};
+
+/**
+ * A hook to get and set the URL search params.
+ *
+ * @returns the URL search params as a key:value map, and a URL
+ *          search params setter method
+ */
+export const useSearchParams = (): [object, SetURLSearchParams] => {
+  const [searchParams, internalSetSearchParams] = React.useState(toMap(location.search));
```

Comment:
Question: Do we want the hook to react if the search params are changed outside of the component?

Reply:
I don't think we should over do it, the idea is to remove this code once v6 is used in the console